### PR TITLE
Issue 9: Check for consul binary

### LINKED ISSUE #9: Check for necessary binaries [Makefile]
_originally posted [#306](https://github.com/fredrikhgrelland/vagrant-hashistack/issues/306) by @zhenik_

Some of the commands in `Makefile` require binaries available on `PATH`, such as
- `consul` (to build a local proxy to the container)
- ...

Todo:
- [x] add check for binaries availability

[Example](https://github.com/fredrikhgrelland/terraform-nomad-redash/blob/e8dec7d3c4f1f5f2330a4def49cbaaf04a0bc5a8/Makefile#L12)

### PR DESCRIPTION
**Pull request by claesgill**
Closes #9 

* Added a check for consul binary in the `Makefile`
* Added a check for terraform binary in the `Makefile`

Let me know if we want to check for other binaries.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 ### Added
 
+- Added check for consul binary in the `Makefile`
```
Comment (zhenik) [inline CHANGELOG.md]:
```suggestion
- Added check for consul binary in the `Makefile` #20
```

**Review by zhenik — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (claesgill) [reply CHANGELOG.md]: 🤦

**Review by Neha-Sinha2305 — CHANGES_REQUESTED** [review]
Is this tested? As in have you tried removing the consul-binary and run this code to check if the error message shows up.
My point is we might need to add a **prereq:** tag, otherwise it is dead code which is not called from anywhere.

File: Makefile [hunk 2]
```diff
@@ -9,6 +9,11 @@ export PATH := $(shell pwd)/tmp:$(PATH)
 install:
 	 mkdir -p tmp;(cd tmp; git clone --depth=1 https://github.com/fredrikhgrelland/vagrant-hashistack.git; cd vagrant-hashistack; make install); rm -rf tmp/vagrant-hashistack
 
+check_for_consul_binary:
+ifeq (, $(shell which consule))
```
Comment (Neha-Sinha2305) [inline Makefile]:
```suggestion
ifeq (, $(shell which consul))
```

**Review by Neha-Sinha2305 — CHANGES_REQUESTED** [review]

**Comment by pdmthorsrud** [comment]
> Is this tested? As in have you tried removing the consul-binary and run this code to check if the error message shows up.
> My point is we might need to add a **prereq:** tag, otherwise it is dead code which is not called from anywhere.

From what I can see there are no  targets in the makefile that actually uses consul. My suggestion is we add a section "how to open a proxy to consul-connect enabled services", and document the target in this PR there

File: Makefile (hunk 2 above)
Comment (claesgill) [reply Makefile]: 😱 Forgot to remove after testing. Thanks for finding that!!

**Review by Neha-Sinha2305 — APPROVED** [review]
LGTM :+1:

**Pull request merged**